packages/@o3r/schematics/src/utility/update-pipes.ts

```diff
@@ -0,0 +1,117 @@
+import { Rule, SchematicContext, Tree } from '@angular-devkit/schematics';
+import { basename, dirname, join } from 'node:path';
+import { EOL } from 'node:os';
+import * as ts from 'typescript';
+import { DecoratorWithArg, isDecoratorWithArg } from './ast';
+import { getO3rComponentInfoOrThrowIfNotFound, isNgClassComponent, isNgClassDecorator } from './component';
+import { findFilesInTree } from './loaders';
+
+/** Dictionary of pipes to be updated */
+export type PipeReplacementInfo = Record<string, { new: { name: string; import?: string }; import: string }>;
+
+const applyChanges = (
+  pipeReplacementInfo: PipeReplacementInfo,
+  ngDecorator: DecoratorWithArg,
+  matchers: RegExpMatchArray[],
+  fileWithImports: string,
+  templateFile: string,
+  tree: Tree,
+  context: SchematicContext
+) => {
+  if (ts.isObjectLiteralExpression(ngDecorator.expression.arguments[0])) {
+    const importsProp = ngDecorator.expression.arguments[0].properties.find((prop): prop is ts.PropertyAssignment & { initializer: ts.ArrayLiteralExpression } =>
+      ts.isPropertyAssignment(prop)
+              && ts.isIdentifier(prop.name)
+              && prop.name.text === 'imports'
+              && ts.isArrayLiteralExpression(prop.initializer)
+    );
+    if (importsProp) {
+      matchers.forEach((matcher) => {
+        const pipeName = matcher[1];
+        const importModuleRegexp = new RegExp(`\\b${pipeReplacementInfo[pipeName].import}\\b`, 'g');
+        if (importsProp.initializer.elements.some((element) => importModuleRegexp.test(element.getText()))) {
+          if (pipeReplacementInfo[pipeName].new.import) {
+            tree.overwrite(
+              fileWithImports,
+              tree.readText(fileWithImports).replaceAll(importModuleRegexp, pipeReplacementInfo[pipeName].new.import!)
+            );
+          }
+          tree.overwrite(
+            templateFile,
+            tree.readText(templateFile).replaceAll(new RegExp(`\\|\\s*${pipeName}`, 'g'), `| ${pipeReplacementInfo[pipeName].new.name}`)
+          );
+        } else {
+          context.logger.warn(
+            `Deprecated pipe ${pipeName} was found in ${templateFile} `
+            + `but no associated import in ${fileWithImports}`
+            + `to be able to migrate to the new pipe ${pipeReplacementInfo[pipeName].new.name}.${EOL}`
```

Comment:
Not sure `${EOL}` is needed in the end of the message, it will already be handled by `logger.warm`